src/coreclr/vm/gccover.cpp

```diff
@@ -1816,37 +1816,20 @@ void DoGcStress (PCONTEXT regs, NativeCodeVersion nativeCodeVersion)
         frame.Push(pThread);
     }
 
-    DWORD_PTR retValRegs[2] = { 0 };
+    // The legacy X86 GC encoder does not encode the state of return registers at
+    // call sites, so we must add an extra frame to protect returns.
+#ifdef TARGET_X86
+    DWORD_PTR retValRegs[1] = { 0 };
     UINT  numberOfRegs = 0;
```

Comment:
```suggestion
    DWORD_PTR retValReg = 0;
```
This can simplified since we are only protecting a single value.

Reply:
I think with runtime async the need to protect two returns will be back fairly soon, so i did not simplify to strictly one return.